source/_integrations/pid_thermostat.markdown

```diff
@@ -0,0 +1,141 @@
+
+## Configuration
+{% include integrations/config_flow.md %}
+{% configuration_basic %}
+Name:
+  description: The name that the PID thermostat should have. You can change it again later.
+Heater device entity:
+  description: Heater- or cooler number entity. Typically, the [slow_pwm number entity][slow_pwm] can be used to create a number controlling a switch.
+Heater device entity:
+  description: Temperature sensor, used for input signal.
+Proportional gain factor (Kp):
+  description: Proportional gain factor, directly gaining the error to compensate the fault (Kp).
+Integration factor (Ki):
+  description: Integration factor, reducing offset fault over time (Ki).
+Differential factor (Kd):
+  description: Differential factor, damping the overshoot (Kd).
+Thermostat mode: 
+  description: Select if the thermostat should be a cooler or a heater.
+Duration between controller cycles
+  description: Cycle time that elapses between two controller cycles.
+{% endconfiguration_basic %}
+
+## YAML Configuration
+
+Alternatlively, this integration can be configured and set up manually via YAML
+instead. To enable the PID thermostat entity in your installation, add the
+following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+climate:
+  - platform: pid_thermostat
+    name: Kitchen thermostat
+    heater: number.floor_heater
+    sensor: sensor.kitchen_temperature
+``` 
+
+{% configuration %}
+name:
+  description: Name of the PID thermostat.
+  required: true
+  type: string
+heater:
+  description: `entity_id` for heater- or cooler, must be number device. Typically, the [slow_pwm number entity][slow_pwm] can be used to create a number controlling a switch.
+  required: true
+  type: string
+sensor:
+  description: `entity_id` for temperature sensor.
+  required: true
+  type: string
+kp:
+  description: Proportional gain factor, directly gaining the error to compensate the fault (Kp).
+  required: false
+  default: 100.0
+  type: float
+ki:
+  description: Integration factor, reducing offset fault over time (Ki).
+  required: false
+  default: 0.1
+  type: float
+kd:
+  description: Differential factor, damping the overshoot (Kd).
+  required: false
+  default: 0.0
+  type: float
+ac_mode:
+  description: Thermostat mode. 
+  required: false
+  default: direct
+  type: string ('cool' or 'heat')
+min_temp:
+  description: Minimal temperature setpoint.
+  required: false
+  default: 7
+  type: float
+max_temp:
+  description: Maximal temperature setpoint.
+  required: false
+  default: 35
+  type: float
+cycle_time:
+  description: Cycle time for the internal PID controller.
+  required: false
+  default: {"seconds": 30}
+  type: time_period
+target_temp:
+  description: Target tempererature on startup.
+  required: false
+  default: 19
+  type: float
+initial_hvac_mode:
+  description: Initial hvac mode. 
+  required: false
+  default: off
+  type: string ('off', 'cool' or 'heat')
```

Comment:
```suggestion
  type: string ('off', 'cool', or 'heat')
```

Reply:
I'm probably not the target audience for this text and maybe this is clear to people who know their way around PID and PWM. How do I know what values to choose for kp, ki, kd? kp defaults to 100, yet in your example you went for 150.

Reply:
Let me add the same explainer to this text as for pid_contoller. When used in a thermostat, one usually works with high kp; if the temperature is 1 degC lower than wished, the heater should switch on at full power, because that's already relative much for such an application.